pkg/infrastructure/azure/storage.go

```diff
@@ -206,30 +254,83 @@ func CreatePageBlob(ctx context.Context, in *CreatePageBlobInput) (*CreatePageBl
 		},
 	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get page blob client: %w", err)
+		return "", fmt.Errorf("failed to get page blob client: %w", err)
 	}
 
-	// This is used in terraform, not sure if it matters
-	metadata := make(map[string]*string, 1)
-	metadata["source_uri"] = to.Ptr(in.ImageURL)
+	logrus.Debugf("Creating Page blob and uploading image to it")
+	if in.ImageURL == "" {
+		_, err = pageBlobClient.Create(ctx, in.ImageLength, nil)
+		if err != nil {
+			return "", fmt.Errorf("failed to create page blob with image contents: %w", err)
+		}
+		// This image (example: ignition shim) needs to be uploaded from a local file.
+		err = doUploadPages(ctx, pageBlobClient, in.BootstrapIgnData, in.ImageLength)
+		if err != nil {
+			return "", fmt.Errorf("failed to upload page blob image contents: %w", err)
+		}
+	} else {
+		// This is used in terraform, not sure if it matters
+		metadata := make(map[string]*string, 1)
+		metadata["source_uri"] = to.Ptr(in.ImageURL)
```

Comment:
Wonder if this could just be:
```suggestion
		metadata := map[string]*string{
		    "source_uri": to.Ptr(in.ImageURL),
		}
```